elastic/kibana
# [Asset Manager] services endpoint

## Summary

Closes https://github.com/elastic/kibana/issues/159641

Implements `/assets/services` endpoint that returns service assets found in the configured source (signals or assets indices). Consumer can provide a `parent` query to filter the returned services. While the _assets_ mode supports any kind of parent/depth thanks to its common interface, the _signals_ mode only supports host parent for the moment.

### Testing
1. pull this branch and point it at an oblt-cli created cluster that uses cross-cluster search to read from the edge cluster
2. add the following[1] to your kibana.yml file
3. hit `/api/asset-manager/assets/services?from=<from>&to=<to>&(parent=<host>)?`. services should be returned. Add/remove parent query string to filter services only running on specific host.
4. update `lockedSource: assets` to read from assets generated by implicit collection
5. repeat 3.


[1]
```
xpack.assetManager:
  alphaEnabled: true
  sourceIndices:
    metrics: remote_cluster:metricbeat*,remote_cluster:metrics-*
    logs: remote_cluster:filebeat*,remote_cluster:logs-*
    traces: remote_cluster:traces-*
    serviceMetrics: remote_cluster:metrics-apm*
    serviceLogs: remote_cluster:logs-apm*
  lockedSource: signals
  implicitCollection:
    enabled: true
    interval: 30s
```
